src/utilities.h

```diff
@@ -555,4 +555,30 @@ static inline std::string trimCopy(std::string s)
     return s;
 }
 
+/**
+ * @brief Collect all existing id attributes within the given model.
+ *
+ * @param model The @c ModelPtr to interrogate.
+ *
+ * @return An @c IdList collection of existing ids.
+ */
+IdList listIds(const ModelPtr &model);
+
+/**
+ * @brief Utility class to iterate through components to collect ids.
+ *
+ * @param component The @c ComponentPtr to interrogate.
+ * @param idList @c IdList collection of found ids.
+ */
+void listComponentIds(const ComponentPtr &component, IdList &idList);
```

Comment:
Should be removed since it's only being used in `utilities.cpp`.

Reply:
5b41e7f4